sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -627,9 +633,11 @@ public final void open(String uId) throws Exception {
      */
     @Override
     public final FileResult close() throws Exception {
-      try (WritableByteChannel theChannel = channel) {
-        LOG.debug("Writing footer to {}.", filename);
-        writeFooter();
+      LOG.debug("Writing footer to {}.", filename);
```

Comment:
update comment? probably log before writeFooter, then log again before FinishWrite.

switch back to try-with-resources, to catch the ownership issue above? finishWrite should not close.